docs/workflow/testing/mono/testing.md

```diff
@@ -1,54 +1,104 @@
+--------------------------------------------------
 ```
 
-If you want to run individual tests, execute this command:
+To run all tests, execute that command, adding `wasm` to the end.
 
+### Testing on Android x64:
+Build the runtime tests for Android x64
+```
+$(REPO_ROOT)/src/tests/build.sh -skipstressdependencies -excludemonofailures os Android x64 <Release/Debug>
 ```
-dotnet build /t:RunCoreClrTest /p:CoreClrTest="<TestName>" $(REPO_ROOT)/src/mono/mono.proj
+
+The last few lines of the build log should contain something like this:
 ```
+--------------------------------------------------
+ Example run.sh command
 
-## Running Library Tests
-Running library tests against Mono is straightforward regardless of configuration.  Simply run the following commands:
+ src/tests/run.sh --coreOverlayDir=<repo_root>artifacts/tests/coreclr/Android.x64.Release/Tests/Core_Root --testNativeBinDir=<repo_root>/artifacts/obj/coreclr/Android.x64.Release/tests --testRootDir=<repo_root>/artifacts/tests/coreclr/Android.x64.Release --copyNativeTestBin Release
+--------------------------------------------------
+```
+To run all tests, execute that command, adding `Android` at the end.
 
-1. Build and set the RuntimeFlavor
+### Additional Documents
+For more details about internals of the runtime tests, please refer to the [CoreCLR testing documents](../coreclr)
 
-```bash
-./build.sh /p:RuntimeFlavor=mono
+## Libraries tests
+### Testing on desktop Mono
+Build and run library tests against Mono JIT or interpreter
+```
+$(REPO_ROOT)/dotnet.sh build /t:Test /p:RuntimeFlavor=mono /p:Configuration=<Release/Debug> $(REPO_ROOT)/src/libraries/<library>/tests
 ```
-or on Windows
-```bat
-build.cmd /p:RuntimeFlavor=mono
+Alternatively, you could execute the following command from `$(REPO_ROOT)/src/mono/netcore`
 ```
+make run-tests-corefx-<library>
+```
+For example, the following command is for running System.Runtime tests:
+```
+make run-tests-corefx-System.Runtime
+```
+### Testing on mobile targets and WebAssembly
+Build and run library tests against Webassembly, Android or iOS. See instructions located in [Library testing document folder](../libraries/)
 
-2. cd into the test library of your choice (`cd src/libraries/<library>/tests`)
+# Running the Mono samples
+There are a few convenient samples located in `$(REPO_ROOT)/src/mono/netcore/sample`, which could help you test your program easily with different flavors of Mono or do a sanity check on the build. The samples are configured with a certain mode. Please refer to the Makefile for configuration details. If you would like to work with other mode, you could edit the Makefile.
 
-3. Run the tests
+* To run the desktop Mono sample, cd to `HelloWorld` and execute:
```

Comment:
The bullet points are back :(
Please use sub-headers.

Reply:
They are very short. I thought bullet points were more suitable.